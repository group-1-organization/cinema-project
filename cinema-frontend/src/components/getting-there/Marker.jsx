import { markerStyle } from './MarkerStyles';

const Marker = () => {
    return (
        <div style={markerStyle}>
            QA
        </div>
    )
}

export default Marker
