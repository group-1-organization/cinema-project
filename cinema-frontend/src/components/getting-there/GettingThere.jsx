import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import Image from 'react-bootstrap/Image';
import GoogleMapReact from 'google-map-react';
import Marker from './Marker';

const { apiKey } = require('../../const.json');

const GettingThere = () => {
    return (
        <div>
            <div className="page-header">
                <h1>How to find us</h1>
            </div>
            <br />
            <Container>
                <Row>
                    <Col md={6} className="mt-4">
                        <Card bg="dark">
                            <Card.Img variant="top" src="/images/cinema.jpg" />
                        </Card>
                    </Col>
                    <Col md={6} className="mt-4">
                        <Card bg="dark">
                            <Card.Img variant="top" src="/images/cineworld-outside.jpg" />
                        </Card>
                    </Col>
                </Row>
                <Row>
                    <Col md={4} className="mt-4">
                        <Card bg="dark">
                            <Card.Body>
                                <Card.Title>Address</Card.Title>
                                <Card.Text>
                                    Bentley Bridge Park <br />
                                    Wednesfield Way <br />
                                    Wolverhampton <br />
                                    WV11 1TZ
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                    <Col md={4} className="mt-4">
                        <Card bg="dark">
                            <Card.Body>
                                <Card.Title>Directions</Card.Title>
                                <Card.Subtitle className="mb-2 text-muted">Bus</Card.Subtitle>
                                <Card.Text>
                                    Buses 59, 561, 562, 613 stop near the cinema. There are also the 69, 89, 25 and 28 lines from the bus station.
                                </Card.Text>
                                <Card.Subtitle className="mb-2 text-muted">Train</Card.Subtitle>
                                <Card.Text>
                                    Nearest station Wolverhampton. Take bus from bus station close by.
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                    <Col md={4} className="mt-4">
                        <Card bg="dark">
                            <Card.Body>
                                <Card.Title>Parking</Card.Title>
                                <Card.Text>
                                    Free parking available is available nearby the cinema.
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                </Row>
                <Row>
                    <Col className="mt-4">
                        <Card bg="dark">
                            <Card.Body>
                                <Card.Title>Map</Card.Title>
                                <div style={{ height: '650px', width: '100%' }}>
                                    <GoogleMapReact
                                        bootstrapURLKeys={{ key: apiKey }}
                                        defaultCenter={{
                                            lat: 52.596324,
                                            lng: -2.092680
                                        }}
                                        defaultZoom={17} >
                                        <Marker
                                            lat={52.596324}
                                            lng={-2.092680}
                                            text="QA Marker" />
                                    </GoogleMapReact>
                                </div>
                            </Card.Body>
                        </Card>
                    </Col>
                </Row>
            </Container >
        </div>
    )
}

export default GettingThere
