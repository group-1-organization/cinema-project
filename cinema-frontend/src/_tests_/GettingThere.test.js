import renderer, { act } from 'react-test-renderer';
import GettingThere from '../components/getting-there/GettingThere';
import Marker from '../components/getting-there/Marker';

test('Getting there page matches snapshot', () => {
    const tree = renderer.create(<GettingThere />);

    expect(tree).toMatchSnapshot();
});

test('Marker matches snapshot', () => {
    const tree = renderer.create(<Marker />);

    expect(tree).toMatchSnapshot();
});