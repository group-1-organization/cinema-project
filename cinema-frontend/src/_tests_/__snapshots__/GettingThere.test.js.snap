// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting there page matches snapshot 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      How to find us
    </h1>
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="mt-4 col-md-6"
      >
        <div
          className="card bg-dark"
        >
          <img
            className="card-img-top"
            src="/images/cinema.jpg"
          />
        </div>
      </div>
      <div
        className="mt-4 col-md-6"
      >
        <div
          className="card bg-dark"
        >
          <img
            className="card-img-top"
            src="/images/cineworld-outside.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="mt-4 col-md-4"
      >
        <div
          className="card bg-dark"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Address
            </div>
            <p
              className="card-text"
            >
              Bentley Bridge Park 
              <br />
              Wednesfield Way 
              <br />
              Wolverhampton 
              <br />
              WV11 1TZ
            </p>
          </div>
        </div>
      </div>
      <div
        className="mt-4 col-md-4"
      >
        <div
          className="card bg-dark"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Directions
            </div>
            <div
              className="mb-2 text-muted card-subtitle h6"
            >
              Bus
            </div>
            <p
              className="card-text"
            >
              Buses 59, 561, 562, 613 stop near the cinema. There are also the 69, 89, 25 and 28 lines from the bus station.
            </p>
            <div
              className="mb-2 text-muted card-subtitle h6"
            >
              Train
            </div>
            <p
              className="card-text"
            >
              Nearest station Wolverhampton. Take bus from bus station close by.
            </p>
          </div>
        </div>
      </div>
      <div
        className="mt-4 col-md-4"
      >
        <div
          className="card bg-dark"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Parking
            </div>
            <p
              className="card-text"
            >
              Free parking available is available nearby the cinema.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="mt-4 col"
      >
        <div
          className="card bg-dark"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Map
            </div>
            <div
              style={
                Object {
                  "height": "650px",
                  "width": "100%",
                }
              }
            >
              <div
                onClick={[Function]}
                onMouseDownCapture={[Function]}
                onMouseMove={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "margin": 0,
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                />
                <div
                  style={
                    Object {
                      "height": "50%",
                      "left": "50%",
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": "50%",
                      "width": "50%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "height": "100%",
                        "left": 0,
                        "margin": 0,
                        "padding": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "height": 0,
                          "left": 0,
                          "position": "absolute",
                          "top": 0,
                          "width": 0,
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "backgroundColor": "white",
                            "border": "5px solid #f44336",
                            "borderRadius": 40,
                            "color": "#3f51b5",
                            "fontSize": 16,
                            "fontWeight": "bold",
                            "height": 40,
                            "left": -20,
                            "padding": 4,
                            "position": "absolute",
                            "textAlign": "center",
                            "top": -20,
                            "width": 40,
                          }
                        }
                      >
                        QA
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Marker matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "5px solid #f44336",
      "borderRadius": 40,
      "color": "#3f51b5",
      "fontSize": 16,
      "fontWeight": "bold",
      "height": 40,
      "left": -20,
      "padding": 4,
      "position": "absolute",
      "textAlign": "center",
      "top": -20,
      "width": 40,
    }
  }
>
  QA
</div>
`;
